website/docs/cdktf/feature-flags.mdx

```diff
@@ -0,0 +1,34 @@
+---
+page_title: Feature Flags
+description: >-
+  You can use feature flags to enable or disable features that effect how code is synthesized.
+---
+
+# Feature Flags
+
+You can use feature flags to enable or disable features that effect how code is synthesized.
+They can be configured in the `cdktf.json` file:
+
+```json
+{
+  "language": "java",
+  "app": "mvn -e -q compile exec:java",
+  "terraformProviders": ["aws@~> 3.0"],
+  "context": {
+    "excludeStackIdFromLogicalIds": "true",
+    "allowSepCharsInLogicalIds": "true"
+  }
+}
+```
+
+All feature flags are disabled by default.
+
+## `excludeStackIdFromLogicalIds`
+
+## `allowSepCharsInLogicalIds`
+
+## Lifecycle
+
+Feature flags can only be used to change the logic when synthesizing code or CLI features. Since we have prebuilt providers we can not change the logic of the provider generation.
+
+A feature flag will be introduced in a release, adding it with a default `false` value to the templates used in `cdktf init`. The feature flag will graduate to be on by default in the next minor release, we will print a warning if it's false or not set. The feature flag will be removed in the next minor release.
```

Comment:
> in the next minor release

Wouldn't this be major? I think semver denotes that differently below 1.0.0 🤔

In my mind we'd graduate them e.g. in the following version jumps
```
0.11.0 -> 0.12.0
1.0.0 -> 2.0.0
```
(which would be major version to me)

... just checked
> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

So, [semver](https://semver.org/spec/v1.0.0.html) does not denote it like I thought 😁

Reply:
Thanks for checking. I was thinking once we hit 1.0 we can `:%s/minor/major/g` here

Reply:
Yeah, that would be fine with me 👍